### PR TITLE
bucket init from Cid, add IPFS UnixFS in local bucket, SetHead API, update Powergate

### LINKED ISSUE #188: Push existing IPFS cid to a bucket
Two scenarios: 

1. Push a new CID to the HEAD of a Bucket.
2. Push a new CID into the path of a Bucket.

The second one would add a lot of flexibility for existing projects that use IPFS to include those hashes in the persistence layer offered by Buckets.

### PR DESCRIPTION
**Pull request by jsign**
This PR is adding three features:
- A new flag for init-ing a bucket with `buck init --cid`, which will create a bucket with bootstrap data from the provided `<cid>`. This Cid is a UnixFS DAG in the IPFS network.
- Adding content to an existing bucket from the IPFS network. `buck add <cid> <path>`.
- A new API for setting an IPFS UnixFS DAG at an arbitrary path in an existing bucket.

Also, it updates to newest version of Powergate and lotus-devnet.

Close #188

### CODE REVIEW

File: api/buckets/service.go [hunk 1]
```diff
@@ -119,6 +132,40 @@ func (s *Service) createBucket(ctx context.Context, dbID thread.ID, dbToken thre
...
+	}
+	buckPath, err := s.IPFSClient.Object().AddLink(ctx, path.IpfsPath(bootCid), bucks.SeedName, seedPth)
+	if err != nil {
+		return nil, fmt.Errorf("adding seed to prepared bucket: %s", err)
+	}
+	if err = s.IPFSClient.Pin().Add(ctx, buckPath); err != nil {
+		return nil, fmt.Errorf("pinning prepared bucket: %s", err)
+	}
+	return buckPath, nil
+}
```
[the remark this answers is not in the document]
Comment (sanderpick) [reply api/buckets/service.go:167]: 👍

File: api/buckets/service.go [hunk 2]
```diff
@@ -178,6 +225,62 @@ func (s *Service) createLinks(dbID thread.ID, buck *bucks.Bucket) *pb.LinksReply
...
+	var dirpth path.Path
+	if req.Path == "" {
+		seed := make([]byte, 32)
+		if _, err := rand.Read(seed); err != nil {
+			return nil, fmt.Errorf("generating new seed: %s", err)
+		}
+		dirpth, err = s.createBootstrappedPath(ctx, seed, remoteCid)
+		if err != nil {
+			return nil, fmt.Errorf("generating bucket new root: %s", err)
+		}
```
[the remark this answers is not in the document]
Comment (sanderpick) [reply api/buckets/service.go:258]: Nice, nope can't think of a reason to keep the original seed.

File: cmd/buck/cli/add.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+	"github.com/textileio/textile/api/buckets/client"
+	"github.com/textileio/textile/cmd"
+)
+
+const (
+	promptIgnore  = "Ignore"
+	promptMerge   = "Merge"
+	promptReplace = "Replace"
```
[the remark this answers is not in the document]
Comment (sanderpick) [reply cmd/buck/cli/add.go:21]: Maybe "Ignore" -> "Skip"? Just a thought

File: cmd/buck/cli/init.go [hunk 4]
```diff
@@ -51,11 +54,19 @@ Use the '--existing' flag to initialize from an existing remote bucket.
...
+			cmd.Fatal(err)
+		}
 		existing, err := c.Flags().GetBool("existing")
 		if err != nil {
 			cmd.Fatal(err)
 		}
+
+		if bootCid != "" && existing {
+			cmd.Fatal(errors.New("only one of --cid and --existing flags can be used at the same time"))
+		}
```
[the remark this answers is not in the document]
Comment (sanderpick) [reply cmd/buck/cli/init.go:68]: Yep, looks good! I can't think of any other problematic combos.

File: go.mod [hunk 5]
```diff
@@ -60,17 +55,18 @@ require (
...
-	go.uber.org/zap v1.15.0 // indirect
-	golang.org/x/net v0.0.0-20200425230154-ff2c4b7c35a0 // indirect
-	golang.org/x/sys v0.0.0-20200519105757-fe76b779f299 // indirect
 	golang.org/x/time v0.0.0-20200416051211-89c76fbcd5d1 // indirect
 	google.golang.org/grpc v1.29.1
 	gopkg.in/ini.v1 v1.55.0 // indirect
 )
+
+// Fixes races. Keep this until this gets tagged and propagated to other deps.
+replace github.com/libp2p/go-libp2p-peerstore => github.com/libp2p/go-libp2p-peerstore v0.2.5-0.20200605182041-9827ee08601f
```
[the remark this answers is not in the document]
Comment (sanderpick) [reply go.mod:71]: Sweet

File: api/buckets/pb/buckets.proto [hunk 6]
```diff
@@ -54,17 +55,25 @@ message ListPathRequest {
...
 
-    message Item {
-        string cid = 1;
-        string name = 2;
-        string path = 3;
-        int64 size = 4;
-        bool isDir = 5;
-        repeated Item items = 6;
-    }
+message ListPathItem {
```
[the remark this answers is not in the document]
Comment (jsign) [reply api/buckets/pb/buckets.proto:62]: Cool, so good time to go for it. :)

File: cmd/buck/cli/add.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsign) [reply cmd/buck/cli/add.go:21]: Oh, I like _Skip_ more sounds closer to a _process_ which is this case. Will change.

**Pull request merged**